### ISSUE TITLE
pwa-chrome launch type disregards port for ui browserLaunchLocation

### ISSUE THREAD

**Issue by lucono** [microsoft/vscode-js-debug#640]
**Describe the bug**
Using the "pwa-chrome" type, "launch" request and "ui" browserLaunchLocation, with "port" set to a non-zero value such as 9222, the port configuration is ignored and Chrome is launched with the `--remote-debugging-pipe` flag instead of `--remote-debugging-port=<specified_port>`.

**To Reproduce**
1. In a new or existing project, create a launch configuration with the following details:
```json
{
    "name": "Launch Chrome with debug port 9222",
    "type": "pwa-chrome",
    "request": "launch",
    "browserLaunchLocation": "ui",
    "url": "http://localhost:8080",
    "port": 9222
}
```
2. Run this launch configuration

**Expected**
- Chrome is executed on the UI side with the `--remote-debugging-port=9222` flag so that it's listening on port 9222 for debug connections.

**Actual**
- Chrome is executed on the UI side with the `--remote-debugging-pipe` flag and not `--remote-debugging-port=9222`.
- Confirmed this by replacing the Chrome executable with a script that logs its arguments, which revealed vscode invoking it with `--remote-debugging-pipe` flag and not `--remote-debugging-port=9222`
- This behavior is the same for both projects opened in a remote "workspace" context as well as local projects in a "ui" only context.

**VS Code Version:** 1.47.2
**OS and Version:** macOS Catalina / 10.15.5

**Comment by lucono** [comment]
@connor4312 

Testing this using the latest [vscode-js-debug](https://marketplace.visualstudio.com/items?itemName=ms-vscode.js-debug-nightly) nightly and the accompanying [js-debug-companion](https://marketplace.visualstudio.com/items?itemName=ms-vscode.js-debug-companion), I see that:
* The `port` key in the launch configuration still seems to have no effect in adding the required `--remote-debugging-port=<configured_port>` command-line flag when launching Chrome.
* I see new behavior where the `--remote-debugging-pipe` command line flag is now no longer always present when launching the Chrome executable, regardless of whether the launch configuration contains the `port` option or not.
* Also observed new behavior where `--remote-debugging-port=` (without a port value after the `=`) is included as a command line flag to the Chrome runtime executable if using the `runtimeArgs` launch configuration option to manually force adding `--remote-debugging-port=9222` (or some other port value) to the execution of the Chrome runtime executable, while also additionally appending again the full `--remote-debugging-port=9222` flag.

Basically:

With launch configuration:
```json
{
    "name": "Launch Chrome with debug port 9222",
    "type": "pwa-chrome",
    "request": "launch",
    "browserLaunchLocation": "ui",
    "url": "http://localhost:8080",
    "port": 9222
}
```
the Chrome binary is executed with the following flags (showing relevant flags only):
```
--user-data-dir=<some_path>
```
And with launch configuration:
```json
{
    "name": "Launch Chrome with debug port 9222",
    "type": "pwa-chrome",
    "request": "launch",
    "browserLaunchLocation": "ui",
    "url": "http://localhost:8080",
    "runtimeArgs": [ "--remote-debugging-port=9222" ]
}
```
then the Chrome binary execution flags are (relevant flags only):
```
--remote-debugging-port= --user-data-dir=<some_path> --remote-debugging-port=9222
```

**Comment by connor4312** [comment]
I'm not able to repro any of the problems you describe. Can you verify that:

1. You have js-debug-companion@1.0.4 installed in your _ui_ locally
1. You can js-debug-nightly@2020.8.1217 install in your _remote_ workspace?

**Comment by lucono** [comment]
@connor4312 

Seems I've got the right versions installed, but also still had the built-in version enabled. Disabling that and using this launch config:
```json
{
    "name": "Launch Chrome with debug port 9222",
    "type": "pwa-chrome",
    "request": "launch",
    "browserLaunchLocation": "ui",
    "url": "http://localhost:8080",
    "port": 9222
}
```
gave me this:
```
--remote-debugging-port= --remote-debugging-port=9222
```
Though this uses the flag twice, in my tests it is able to launch Chrome and attach on that port with no issues. So I think functionally this looks good - Thanks!
